go/libraries/doltcore/mvdata/data_mover.go

```diff
@@ -97,7 +100,7 @@ func NewDataMover(ctx context.Context, root *doltdb.RootValue, fs filesys.Filesy
 	var err error
 	transforms := pipeline.NewTransformCollection()
 
-	rd, srcIsSorted, err := mvOpts.Src.NewReader(ctx, root, fs, mvOpts.SchFile, mvOpts.SrcOptions)
```

Comment:
Added `VerifySchema` method to `TableReader` so that verification only happens in [this block](https://github.com/liquidata-inc/dolt/pull/86/files#diff-1b5d76ac2f2d89a0d1c3837d97f0b7f3R123)